### PR TITLE
feat(DC): Fast JSON Schema

### PR DESCRIPTION
**Pull request by rahul-kumar-saini**
### Overview
- The true bottleneck for Snuba init times is the JSON Schema validation
- Apparently a very common problem with plenty of solutions:
     - https://www.peterbe.com/plog/jsonschema-validate-10x-faster-in-python
- Locally I'm seeing a solid 50% reduction in load times by compiling validators once and reusing them across the configs
- Each new config should add very little to the init time as well since the bulk of the time will be spent compiling the validators in the beginning

### Previously, each time we validate a config the schema was compiled and validated against:
Screen Shot 2022-11-22 at 11 38 57 AM

### Now we compile the validators beforehand and just run them against each config:
Screen Shot 2022-11-22 at 12 01 33 PM

### Notes
- I removed the `build` span from everything since building the actual object takes <1ms

### CODE REVIEW

**Comment by enochtangg** [comment]
LGTM!

File: docs-requirements.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
 jsonschema2md==0.4.0
+fastjsonschema==2.16.2
```
Comment (evanh) [inline docs-requirements.txt:2]: Why do you need these in docs?

**Review by evanh — COMMENTED** [review]
Why did you remove all the spans around building the Python objects (i.e. `description=f"Storage: {config['name']}")`?

**Comment by rahul-kumar-saini** [comment]
> Why did you remove all the spans around building the Python objects (i.e. `description=f"Storage: {config['name']}")`?

I removed the `build` span from everything since building the actual object takes <1ms

I did add back the info of what storage is being loaded+validated in the description:

image

File: docs-requirements.txt (hunk 1 above)
Comment (rahul-kumar-saini) [reply docs-requirements.txt:2]: Docs use the `json_schema.py` file to generate the docs, I moved the compilation of the schemas into validators into that file. All of that files dependencies need to be installed to run the docs command, I could move it out to another new file if this isn't a good solution.

**Review by lynnagara — APPROVED** [review]
Seems good to me

**Pull request merged**